persistence/provider/boltdb/transaction.go

```diff
@@ -11,27 +11,33 @@ import (
 // data stores.
 type transaction struct {
 	ds     *dataStore
+	result *persistence.TransactionResult
```

Comment:
```suggestion
	result persistence.TransactionResult
```

Regardless of whether we choose to retain the pointer return value from `Commit()`, these `result` fields don't need to be a pointer - they're always needed in the happy path.